test/integration/discovery.go

```diff
@@ -309,16 +314,100 @@ var _ = Describe("Discover Cluster", func() {
+
+			//Verify zone label in the commodity list
+			podEntityDTO, nodeEntityDTO = getPodNodeDTO(entityDTOs, podFullName)
+
+			// Verify zone label is honored for pod with pv
+			if !validateBuyerSellerCommodity(podEntityDTO, nodeEntityDTO, nodeName, proto.CommodityDTO_LABEL, commodityZoneValue) {
+				framework.Failf("Zone label in node for pod with PV is not honored")
+			}
+			if !validateBuyerSellerCommodity(podEntityDTO, nodeEntityDTO, nodeName, proto.CommodityDTO_LABEL, commodityRegionValue) {
+				framework.Failf("Region label in node for pod with PV is not honored")
+			}
+
+		})
+
+		It("discovering with pv with affinity rules with featureGate disabled", func() {
+
+			// Disable  the feature gate
+			honorAzPvFlag := make(map[string]bool)
+			honorAzPvFlag["HonorAzLabelPvAffinity"] = false
+			err = utilfeature.DefaultMutableFeatureGate.SetFromMap(honorAzPvFlag)
+			if err != nil {
+				glog.Fatalf("Invalid Feature Gates: %v", err)
+			}
 
 			entityDTOs, _, err := discoveryClient.DiscoverWithNewFramework("discovery-integration-test")
 			framework.ExpectNoError(err, "Failed completing discovery of test cluster")
+			podEntityDTO, nodeEntityDTO = getPodNodeDTO(entityDTOs, podFullName)
 
-			//Validation logic
-			if !validateBuyerSellerCommodity(entityDTOs, nodeNameForPod, podFullName, proto.CommodityDTO_VMPM_ACCESS, "foo in (bar)") {
+			var delNode *corev1.Node = getNodewithNodeName(nodeName, kubeClient)
+			if validateBuyerSellerCommodity(podEntityDTO, nodeEntityDTO, nodeName, proto.CommodityDTO_VMPM_ACCESS, "foo in (bar)") {
+				podNode = deleteLabelsFromNode(delNode, "foo", kubeClient)
 				framework.Failf("PV affinity is not honored")
 			}
+			//Manual deletion of foo=bar
+			podNode = deleteLabelsFromNode(delNode, "foo", kubeClient)
+		})
+
+		It("pod with pv to honor zone label when feature gate is default(false)", func() {
+
+			//Use Case 3 : Test zone label in node and featureGate value false
+
+			// Disable the feature gate
+			honorAzPvFlag := make(map[string]bool)
+			honorAzPvFlag["HonorAzLabelPvAffinity"] = false
+			err = utilfeature.DefaultMutableFeatureGate.SetFromMap(honorAzPvFlag)
+			if err != nil {
+				glog.Fatalf("Invalid Feature Gates: %v", err)
+			}
+
+			podNode, err = updateNodeWithLabel(kubeClient, zoneLabelKey, zoneLabelValue, nodeName)
+			podNode, err = updateNodeWithLabel(kubeClient, regionLabelKey, regionLabelValue, nodeName)
+
+			entityDTOs, _, err := discoveryClient.DiscoverWithNewFramework("discovery-integration-test")
+			framework.ExpectNoError(err, "Failed completing discovery of test cluster")
+
+			podEntityDTO, nodeEntityDTO = getPodNodeDTO(entityDTOs, podFullName)
+
+			//Verify zone/region label in the commodity list
+			if validateBuyerSellerCommodity(podEntityDTO, nodeEntityDTO, nodeName, proto.CommodityDTO_LABEL, commodityZoneValue) {
+				framework.Failf("Zone label in node for pod with PV is not honored")
```

Comment:
I think the fail message  should be `Zone label in node for pod with PV is honored` in this case